index.js

```diff
@@ -480,90 +489,88 @@ RingPop.prototype.issueMembershipChanges = function issueMembershipChanges(check
     return this.dissemination.getChanges(checksum, source);
 };
 
+RingPop.prototype.onMemberAlive = function onMemberAlive(member) {
+    this.stat('increment', 'membership-update.alive');
+
+    this.dissemination.addChange({
+        address: member.address,
+        status: member.status,
+        incarnationNumber: member.incarnationNumber,
+        piggybackCount: 0
+    });
+
+    this.ring.addServer(member.address);
+    this.suspicion.stop(member);
+};
+
+RingPop.prototype.onMemberFaulty = function onMemberFaulty(member) {
+    this.stat('increment', 'membership-update.faulty');
+
+    this.dissemination.addChange({
+        address: member.address,
+        status: member.status,
+        incarnationNumber: member.incarnationNumber,
+        piggybackCount: 0
+    });
+
+    this.ring.removeServer(member.address);
+    this.suspicion.stop(member);
+};
+
+RingPop.prototype.onMemberJoined = function onMemberJoined(member) {
+    this.stat('increment', 'membership-update.new');
+
+    this.ring.addServer(member.address);
+
+    this.dissemination.addChange({
+        address: member.address,
+        status: member.status,
+        incarnationNumber: member.incarnationNumber,
+        piggybackCount: 0
+    });
+};
+
+RingPop.prototype.onMemberLeave = function onMemberLeave(member) {
+    this.stat('increment', 'membership-update.leave');
+
+    this.dissemination.addChange({
+        address: member.address,
+        status: member.status,
+        incarnationNumber: member.incarnationNumber,
+        piggybackCount: 0
+    });
+
+    this.ring.removeServer(member.address);
+    this.suspicion.stop(member);
+};
+
+RingPop.prototype.onMemberSuspect = function onMemberSuspect(member) {
+    this.stat('increment', 'membership-update.suspect');
+
+    this.suspicion.start(member);
+
+    this.dissemination.addChange({
+        address: member.address,
+        status: member.status,
+        incarnationNumber: member.incarnationNumber,
+        piggybackCount: 0
+    });
+};
+
 RingPop.prototype.onMembershipUpdated = function onMembershipUpdated(updates) {
     var self = this;
 
-    var updateHandlers = {
-        alive: function onAliveMember(member) {
-            /* jshint camelcase: false */
-            self.stat('increment', 'membership-update.alive');
-            self.logger.info('member is alive', {
```

Comment:
Your allowed to downgrade this to a `.trace()`.

The latests logtron configures the trace level to log nowhere by default and must be opt into by the application developer.

This allows you to leave very verbose logging in place that could be turned on with a redeploy of the app config.